### PR TITLE
Chore(E2E): Adding E2E test of resource table widget

### PR DESCRIPTION
**Pull request by MedWassCh**
## Description

**PLEASE MAKE SURE THAT THE BRANCH PR INCLUDES JIRA TICKET ID** (_for centreon-internal_)

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/tests/e2e/features/Dashboards/13-resource-table-widget-configuration.feature [hunk 1]
```diff
@@ -0,0 +1,55 @@
+@REQ_MON-24518
+Feature: Configuring resource table widget
+  As a Centreon User with dashboard update rights,
+  I need to configure a widget containing a resource table on a dashboard
+  To manipulate the properties of the resource table Widget and test the outcome of each manipulation.
+
+  @TEST_MON-35094
+  Scenario: Editing the displayed resource status of ressource table widget
```
Comment (paul-oureib) [inline centreon/tests/e2e/features/Dashboards/13-resource-table-widget-configuration.feature:8]: please correct every instance of "ressource" with "resource"

File: centreon/tests/e2e/features/Dashboards/13-resource-table-widget-configuration.feature [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  @TEST_MON-35096
+  Scenario: Displaying ressources on a downtime on a ressource table widget
+    Given a dashboard featuring a configured ressource table widget
+    When the dashboard administrator user applies a downtime on one of the resources of the dataset selection
+    And selects the option to exclusively display the resources on a downtime
+    Then only the resource on a downtime is displayed in the ressrouce table widget
+
+  @TEST_MON-35098
+  Scenario: Duplicating ressource table widget
```
Comment (paul-oureib) [inline centreon/tests/e2e/features/Dashboards/13-resource-table-widget-configuration.feature:21]:
```suggestion
  Scenario: Duplicating a resource table widget
```

File: centreon/tests/e2e/features/Dashboards/13-resource-table-widget-configuration.feature [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    Given a dashboard having a configured ressrouce table widget
+    When the dashboard administrator user duplicates the ressource table widget
+    Then a second ressrouce table widget is displayed on the dashboard
```
Comment (paul-oureib) [inline centreon/tests/e2e/features/Dashboards/13-resource-table-widget-configuration.feature:24]:
```suggestion
    Then a second resource table widget is displayed on the dashboard
```

File: centreon/tests/e2e/features/Dashboards/13-resource-table-widget-configuration.feature [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    Given a dashboard that includes a configured ressource table widget
+    When the dashboard administrator user selects a particular status in the displayed ressrouce state list
+    Then only the resources with this particular state are displayed in the ressource table Widget
+
+  @TEST_MON-35097
+  Scenario: Displaying aknowledged ressources on a ressource table widget
+    Given a dashboard featuring a configured ressrouce table widget
+    When the dashboard administrator user applies an acknowledgement on one of the resources of the dataset selection
+    And selects the option to exclusively display the acknowledged resources
+    Then only the acknowledged resource is displayed in the ressrouce table widget
```
Comment (paul-oureib) [inline centreon/tests/e2e/features/Dashboards/13-resource-table-widget-configuration.feature:55]:
```suggestion
    Then only the acknowledged resource is displayed in the resource table widget
```

File: centreon/packages/js-config/cypress/e2e/configuration.ts [hunk 5]
```diff
@@ -61,7 +61,7 @@ export default ({
     env: {
       ...env,
       OPENID_IMAGE_VERSION: process.env.MAJOR || '24.04',
-      WEB_IMAGE_OS: 'alma9',
+      WEB_IMAGE_OS: 'bookworm',
```
Comment (kduret) [inline centreon/packages/js-config/cypress/e2e/configuration.ts:64]: restore alma9

File: centreon/tests/e2e/features/Dashboards/13-resource-table-widget-configuration/index.ts [hunk 6]
```diff
@@ -0,0 +1,547 @@
...
+  }).realClick();
+  cy.wait('@resourceRequest');
+});
+
+When(
+  'the dashboard administrator user selects view by host as a display type',
+  () => {
+    cy.get('svg[data-icon="View by host"]')
+      .should('exist')
+      .click({ force: true });
```
Comment (kduret) [inline centreon/tests/e2e/features/Dashboards/13-resource-table-widget-configuration/index.ts:223]: you should use realClick instead forcing click ?

File: centreon/tests/e2e/features/Dashboards/commands.ts [hunk 7]
```diff
@@ -116,6 +116,25 @@ Cypress.Commands.add(
   }
 );
 
+Cypress.Commands.add('getCellContent', (rowIndex, columnIndex) => {
+  // Utilisez cy.waitUntil pour attendre que l'élément soit visible
```
Comment (kduret) [inline centreon/tests/e2e/features/Dashboards/commands.ts:120]: translate to english

File: centreon/tests/e2e/features/Dashboards/commands.ts [hunk 8]
```diff
@@ -116,6 +116,25 @@ Cypress.Commands.add(
...
+  cy.waitUntil(() =>
+    cy.get(`.MuiTable-root:eq(1) .MuiTableRow-root:nth-child(${rowIndex}) .MuiTableCell-root:nth-child(${columnIndex})`)
+      .should('be.visible')
+      .then(() => true),
+    { timeout: 10000, interval: 1000 }
+  );
+  // Maintenant, l'élément est visible, vous pouvez procéder avec invoke('text')
```
Comment (kduret) [inline centreon/tests/e2e/features/Dashboards/commands.ts:127]: translate to english

File: centreon/tests/e2e/features/Dashboards/commands.ts [hunk 9]
```diff
@@ -116,6 +116,25 @@ Cypress.Commands.add(
...
+  return cy.get(`.MuiTable-root:eq(1) .MuiTableRow-root:nth-child(${rowIndex}) .MuiTableCell-root:nth-child(${columnIndex})`)
+    .invoke('text')
+    .then((content) => {
+      const columnContents = content ? content.match(/[A-Z][a-z]*/g) || [] : [];
+      cy.log(`Contenu de la colonne (${rowIndex}, ${columnIndex}): ${columnContents.join(',').trim()}`);
```
Comment (kduret) [inline centreon/tests/e2e/features/Dashboards/commands.ts:132]: translate to english

File: centreon/tests/e2e/features/Dashboards/14-resource-table-widget-configuration.feature [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+  @TEST_MON-35094
+  Scenario: Editing the displayed resource status of resource table widget
+    Given a dashboard containing a configured resource table widget
+    When the dashboard administrator user selects a particular status in the displayed resource status list
+    Then only the resources with this particular status are displayed in the resource table Widget
+    When the dashboard administrator user selects all the status and save changes
+    Then all the resources having the status selected are displayed in the resource table Widget
+
+  @TEST_MON-35096
+  Scenario: Displaying unhandled ressources on a resource table widget
```
Comment (bdauria) [inline centreon/tests/e2e/features/Dashboards/14-resource-table-widget-configuration.feature:24]:
```suggestion
  Scenario: Displaying unhandled resources on a resource table widget
```

This should be applied in the other parts of the feature

File: centreon/tests/e2e/features/Dashboards/14-resource-table-widget-configuration/index.ts [hunk 11]
```diff
@@ -0,0 +1,603 @@
...
+    () =>
+      cy
+        .get(
+          `.MuiTable-root .MuiTableRow-root:nth-child(1) .MuiTableCell-root:nth-child(1)`
+        )
+        .should('be.visible')
+        .invoke('text')
+        .then((content) => {
+          const columnContents: Array<string> =
+            content.match(/[A-Z][a-z]*/g) || [];
```
Comment (kduret) [inline centreon/tests/e2e/features/Dashboards/14-resource-table-widget-configuration/index.ts:245]:
```suggestion
            content.match(/\w+/g) || [];
```

File: centreon/tests/e2e/features/Dashboards/14-resource-table-widget-configuration/index.ts [hunk 12]
```diff
@@ -0,0 +1,603 @@
...
```
Comment (kduret) [inline centreon/tests/e2e/features/Dashboards/14-resource-table-widget-configuration/index.ts:272]:
```suggestion
            content.match(/\w+/g) || [];
```

File: centreon/tests/e2e/features/Dashboards/14-resource-table-widget-configuration/index.ts [hunk 13]
```diff
@@ -0,0 +1,603 @@
...
+      () =>
+        cy
+          .get(
+            `.MuiTable-root .MuiTableRow-root:nth-child(1) .MuiTableCell-root:nth-child(1)`
+          )
+          .should('be.visible')
+          .invoke('text')
+          .then((content) => {
+            const columnContents: Array<string> =
+              content.match(/[A-Z][a-z]*/g) || [];
```
Comment (kduret) [inline centreon/tests/e2e/features/Dashboards/14-resource-table-widget-configuration/index.ts:327]:
```suggestion
              content.match(/\w+/g) || [];
```

File: centreon/tests/e2e/features/Dashboards/14-resource-table-widget-configuration/index.ts [hunk 14]
```diff
@@ -0,0 +1,603 @@
...
```
Comment (kduret) [inline centreon/tests/e2e/features/Dashboards/14-resource-table-widget-configuration/index.ts:404]:
```suggestion
              content.match(/\w+/g) || [];
```

File: centreon/tests/e2e/features/Dashboards/14-resource-table-widget-configuration/index.ts [hunk 15]
```diff
@@ -0,0 +1,603 @@
...
+        .then(() => true),
+    { interval: 1000, timeout: 10000 }
+  );
+  cy.get(
+    `.MuiTable-root .MuiTableRow-root:nth-child(1) .MuiTableCell-root:nth-child(1)`
+  )
+    .should('be.visible')
+    .invoke('text')
+    .then((content) => {
+      const columnContents = content.match(/[A-Z][a-z]*/g) || [];
```
Comment (kduret) [inline centreon/tests/e2e/features/Dashboards/14-resource-table-widget-configuration/index.ts:458]:
```suggestion
      const columnContents = content.match(/\w+/g) || [];
```

File: centreon/tests/e2e/features/Dashboards/14-resource-table-widget-configuration/index.ts [hunk 16]
```diff
@@ -0,0 +1,603 @@
...
+      () =>
+        cy
+          .get(
+            `.MuiTable-root:eq(1) .MuiTableRow-root:nth-child(1) .MuiTableCell-root:nth-child(1)`
+          )
+          .should('exist')
+          .invoke('text')
+          .then((content) => {
+            const columnContents: Array<string> =
+              content.match(/[A-Z][a-z]*/g) || [];
```
Comment (kduret) [inline centreon/tests/e2e/features/Dashboards/14-resource-table-widget-configuration/index.ts:503]:
```suggestion
              content.match(/\w+/g) || [];
```

File: centreon/tests/e2e/features/Dashboards/14-resource-table-widget-configuration/index.ts [hunk 17]
```diff
@@ -0,0 +1,603 @@
...
+    () =>
+      cy
+        .get(
+          `.MuiTable-root .MuiTableRow-root:nth-child(1) .MuiTableCell-root:nth-child(1)`
+        )
+        .should('be.visible')
+        .invoke('text')
+        .then((content) => {
+          const columnContents: Array<string> =
+            content.match(/[A-Z][a-z]*/g) || [];
```
Comment (kduret) [inline centreon/tests/e2e/features/Dashboards/14-resource-table-widget-configuration/index.ts:593]:
```suggestion
            content.match(/\w+/g) || [];
```

File: centreon/tests/e2e/features/Dashboards/commands.ts [hunk 18]
```diff
@@ -116,6 +116,35 @@ Cypress.Commands.add(
...
+    { interval: 1000, timeout: 10000 }
+  );
+
+  return cy
+    .get(
+      `.MuiTable-root:eq(1) .MuiTableRow-root:nth-child(${rowIndex}) .MuiTableCell-root:nth-child(${columnIndex})`
+    )
+    .invoke('text')
+    .then((content) => {
+      const columnContents = content ? content.match(/[A-Z][a-z]*/g) || [] : [];
```
Comment (kduret) [inline centreon/tests/e2e/features/Dashboards/commands.ts:137]:
```suggestion
      const columnContents = content ? content.match(/\w+/g) || [] : [];
```

**Review by kduret — APPROVED** [review]

**Review by bdauria — APPROVED** [review]

**Pull request merged**